api/v1alpha1/supergraph_types.go

```diff
@@ -0,0 +1,56 @@
+/*
+
+
+Licensed under the Mozilla Public License (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.mozilla.org/en-US/MPL/2.0/
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SuperGraphSpec defines the desired state of SuperGraph
+type SuperGraphSpec struct {
```

Comment:
Move it in model folder. 
Also name it as SuperGraphConfig, like you did with Subgraph, to have consistency or vice-versa.